### ISSUE TITLE
Localization for "en.lproj" has typos which override "Base.lproj"

### ISSUE THREAD

**Issue by svobs** [iina/iina#4468]
- macOS 13.4
- IINA develop branch 39801567ac3c836e655f37041bf2af5f277cad69

**Expected behavior:**

English localizations should match the values in "Base" localization exactly.

**Actual behavior:**

Entries in `iina/en.lproj/Localizable.strings` are missing or incorrect relative to `iina/Base.lproj/Localizable.strings`, but because the "en" translation is present, it will always be used instead of the "Base", resulting in incorrect US English text.

Some examples:

| Correct text  | Actual visible text |
| ------------- | ------------- |
| "Add overlays on videos"  | "Add overlay on videos"  |
| "Error occurred during the installation:"  | "Error occured during the installation:"  |
| "Cannot connect"  | "osd.cannot_connect"  |
| "Timed Out"  | "osd.timed_out"  |

**Steps to reproduce:**
 
As @low-batt [pointed out](https://github.com/iina/iina/pull/4467#pullrequestreview-1484206035):
>The somewhat confusing IINA rules on localization require both the base and english localization files be updated. The rest are updated through [Crowdin](https://translate.iina.io/). So `Localizable.strings` under `en.lproj` also needs to be updated as a part of this PR.
> 
> I've been confused as to why we have to duplicate strings like this. Seems wrong. This PR got me to look into that. From [Enabling Base Internationalization](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/InternationalizingYourUserInterface/InternationalizingYourUserInterface.html#//apple_ref/doc/uid/10000171i-CH3-SW4):
>> Xcode modifies your project folder according to the selections you make in this dialog. Xcode creates a Base.lproj folder in your project folder and adds to it the resource files you select. Xcode creates a language folder for the development language but only adds resources that need translation to the folder. For example, if you select English as the development language, Xcode inserts the resource file in the Base.lproj project folder but not the en.lproj folder because the resource is already in English.
> 
> Checking `info.plist` confirms the development language is set to English:
```xml
<key>CFBundleDevelopmentRegion</key>
<string>en</string>
```
> From the Apple documentation it sounds like `en.lproj` is not needed.

I was able to confirm this by (1) selecting `Localizable.strings` in XCode's Navigator, and (2) deselecting `English`. When the confirmation dialog comes up, check the box to also delete the file on disk.
![SCR-20230616-nvjf](https://github.com/iina/iina/assets/2213815/a9944990-9af7-4a79-b9c2-f79239cfce6a)
<img width="297" alt="SCR-20230616-omnj" src="https://github.com/iina/iina/assets/2213815/ab178cc5-8b98-40ce-8476-546286d80d3c">

At this point I was able to confirm that the `iina/en.lproj/Localizable.strings` was deleted, but IINA was still able to run in US English and it used the Base localization.

I set "English (US)" as the Primary language in the MacOS settings by moving it to the first line, like this:
<img width="715" alt="SCR-20230616-ohuo" src="https://github.com/iina/iina/assets/2213815/fe6dcc9e-4463-4e88-8123-8f345d64d03f">

Note: when changing Primary language, it asks you if you want to restart. You do not need to restart for it to take effect. You only need to restart IINA.

By toggling between `English` and `English (UK)` in the MacOS Settings after making alterations to the `iina/en-gb.lproj/Localizable.strings`, I was able to confirm that "Base" localization was indeed being used as the US English localization, and only that localization.

I went back to XCode's File Inspector, and re-enabled the English localization:
<img width="396" alt="SCR-20230616-ouzn" src="https://github.com/iina/iina/assets/2213815/7a908819-956a-4f72-a272-6bd454ec8c08">

At this point XCode created a new `iina/en.lproj/Localizable.strings` which was clearly copied from `iina/Base.lproj/Localizable.strings`, and staged the changes in Git.

By viewing the staged changes, it's apparent which entries are missing/incorrect.

Note that I did not test the bad entries directly, because they're somewhat hard to get to naturally. Instead I tried deleting / changing the entries for some of the menu items because they're much more visible, then running IINA and viewing their behavior. From that it's easy to infer the behavior of the actual bad items.

**Comment by low-batt** [comment]
Thanks for looking into this!

I suspected we might find the two localizations don't match. Another reason to clean this up.

Might want to hold off making changes to the Xcode project at the moment as it still hasn't been updated to match up with the updated dependencies. We are still waiting on @uiryuu to merge that fix from the 1.3.2 branch.

**Comment by svobs** [comment]
> Might want to hold off making changes to the Xcode project at the moment as it still hasn't been updated to match up with the updated dependencies. We are still waiting on @uiryuu to merge that fix from the 1.3.2 branch.

Sure. To be clear, best & easiest fix (though not super intuitive) is just to just deselect `English` from `Localizable.strings` and delete the file as shown in the pictures, and in the future make sure no one re-selects it again by mistake. Would be good to get this patched soon.

**Comment by uiryuu** [comment]
> At this point I was able to confirm that the iina/en.lproj/Localizable.strings was deleted, but IINA was still able to run in US English and it used the Base localization.

IINA indeed doesn't have to rely on `en.lproj` to run in US locale. The reason why we were using both base localization `en.lproj` together is for Crowdin. From the very beginning when we started to use Crowdin as the translation platform, we configured Crowdin to use `en.lproj` as the base localization.

Crowdin seems to be able to extract strings from `.xib` files, at least when we manually upload `.xib` files to its website. If we can find an automatically way to let Crowdin use `.xib` as the base, then I think we don't have to maintain the `en.lproj` anymore. I'll investigate into about this.

**Comment by uiryuu** [comment]
Unfortunetely, Crowdin still [doesn't](https://store.crowdin.com/categories/file-formats) support to parse `.xib` files. So I fixed the issue by disabling and enabling the English localization. Did you find other discrepancies of Base and English other than the Localizable.strings?

**Comment by svobs** [comment]
> Unfortunetely, Crowdin still [doesn't](https://store.crowdin.com/categories/file-formats) support to parse .xib files. So I fixed the issue by disabling and enabling the English localization. 

That's too bad, but it's not crippling. What IINA really needs is for someone(s) to take ownership over the localizations and re-test before every release, because it's too easy to let them fall out of sync. Unfortunately having so many languages to support makes it hard to verify the correctness, but at least one can see if something is missing.

Closing this issue as won't fix.
